### PR TITLE
Add pytest to testing tools

### PR DESCRIPTION
**Pull request by maltesander**
# Description

*Please add a description here.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: testing-tools/python/requirements.txt [hunk 1]
```diff
@@ -11,4 +11,5 @@ trino==0.322.0
 gssapi==1.8.3
 kerberos==1.3.1
 pure-sasl==0.6.2
+pytest==8.0.1
```
Comment (sbernauer) [inline testing-tools/python/requirements.txt:14]:
```suggestion
pytest==8.0.1 # Needed to execute Python unit-tests
```

**Review by sbernauer — DISMISSED** [review]

**Review by sbernauer — APPROVED** [review]

**Comment by razvan** [comment]
Just out of curiosity: what is this needed for ? Isn't the built in unittest module enough?

**Comment by maltesander** [comment]
Good question next question :D

**Comment by sbernauer** [comment]
Chief Python developer has spoken :pray:

**Comment by maltesander** [comment]
I added it because its required in a test script and i wanted to get the test pipeline working, not really thought about it.
Since its just for internal testing, whats against that @razvan ?

**Comment by razvan** [comment]
> I added it because its required in a test script and i wanted to get the test pipeline working, not really thought about it. Since its just for internal testing, whats against that @razvan ?

Pytest is great. I was just wondering if we need something that unittest can't already do.

So, go ahead.

**Pull request merged**